uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/leave-request-api.js

```diff
@@ -61,6 +61,125 @@ define([
           }));
 
         return deferred.promise;
+      },
+
+      /**
+       * This method is used to update a leave request
+       *
+       * @param {object} updatedAttributes - Updated values of leave request
+       * @return {Promise} Resolved with {Object} Updated Leave request
+       */
+      update: function (updatedAttributes) {
+        $log.debug('LeaveRequestAPI.update');
+
+        return this.sendPOST('LeaveRequest', 'create', updatedAttributes);
+      },
+
+      /**
+       * Gets the overall balance change after a leave request is created. The
+       * API will create and return the detailed breakdown of it in days.
+       *
+       * @param {Object} params matched the API end point params like
+       * mandatory values for contact_id, from_date, from_date_type and optional values for
+       * to_date and to_type.
+       *
+       * @return {Promise} containing the detailed breakdown of balance leaves
+       */
+      calculateBalanceChange: function (params) {
+        $log.debug('LeaveRequestAPI.calculateBalanceChange');
+        var deferred = $q.defer();
+
+        if (params && !('contact_id' in params) || !('from_date' in params) || !('from_date_type' in params)) {
+          deferred.resolve({
```

Comment:
This should be a `deferred.reject`. But why are you doing this validation? Isn't this supposed to be taken care of by the backend endpoint already (cc @davialexandre )? Aren't we duplicating validation logic?